### DISCUSSION TITLE
apply=groupby with aggregate does not return @odata.context

### DISCUSSION THREAD

**Discussion by oleg-ua** [OData/AspNetCoreOData#1036] [General]
before asking, i've read this issue: https://github.com/OData/WebApi/issues/1712

performing the request http://localhost/tf/dataapi/workspaces
returns
{
    "@odata.context": "http://localhost/tf/dataapi/$metadata#Workspaces",
    "value": [
        {
            "Label": "W 638271950392162608",
            "Code": "W 638271950392162608",
            "IsActive": true,
            "Description": "Generated by Watin test",
            "IsHidden": false,
            "CreatedOn": "2023-08-09T16:23:59+03:00",
            "LastModifiedOn": "2023-08-09T16:23:59+03:00",
            "Id": 116
        },
...
}

but performing the request 
http://localhost/tf/dataapi/workspaces?apply=groupby((IsActive),aggregate($count as Total))
returns
 [
    {
        "IsActive": true,
        "Total": 309
    },
    {
        "IsActive": false,
        "Total": 1
    }
]

so, in the second request @odata.context is missing and thus the structure of the response is different - only value is returned.

The controller method looks like this: 
`    [EnableQueryTopSkipFilter(AllowedQueryOptions = Select | Expand | Top | OrderBy | Count | Skip | Filter | Apply)]
    [SwaggerResponse(Status200OK, "Objects successfully retrieved.", typeof(IEnumerable<IDataApiEntity>))]
    public virtual async Task<ActionResult<IQueryable<TDataApiEntity>>> Get(ODataQueryOptions<TDataApiEntity> odataQueryOptions)
    {
        return await TryCatchAsync<TDataApiEntity>(async () =>
        {
            PagedCollectResult<TDataApiEntity> entities = await Entities.Get(odataQueryOptions);

            return Page(entities, odataQueryOptions);
        });
    }`

the Page Method:
` protected OkObjectResult Page<TDataApiEntity>(PagedCollectResult<TDataApiEntity> paginatedQueryResult, ODataQueryOptions<TDataApiEntity> odataQueryOptions)
    {
        IReadOnlyList<TDataApiEntity> entities = paginatedQueryResult.Result;

        if (odataQueryOptions.SetTotalCount())
        {
            // Bug: We need this line because to some internal AspNtCore.OData stuff. Just adding TotalCount number to PageResult doesn't work.
            Request.ODataFeature().TotalCountFunc = () => paginatedQueryResult.TotalNumberOfEntities;
        }

        Request.ODataFeature().NextLink = entities?.Count > 0
            ? Request.GetNextPageLink(paginatedQueryResult.TotalNumberOfEntities)
            : null;

        PageResult<TDataApiEntity> value = new(entities?.AsQueryable() ?? Enumerable.Empty<TDataApiEntity>().AsQueryable(),
            Request.ODataFeature().NextLink,
            odataQueryOptions.SetTotalCount() ? Request.ODataFeature().TotalCount.GetValueOrDefault() : null);

        return Ok(value);
    }`
  

So, return type of the Get method is ActionResult<...>, but not IQueryable<...>. I saw in exampleas, that @odata.context is returned in case when controller returns IQueriable<...>.  
Is it possible to return odata.context in my case as well? 
Or maybe you can suggest what I should modify in the EDM model?

**Comment by peterkovecses** [comment]
I had the same issue, I solved it like this:
https://github.com/peterkovecses/ODataApplyDemo

**Comment by valentasm1** [comment]
Couple solutions are provided here
https://github.com/OData/AspNetCoreOData/issues/181
